src/borg/repository.py

```diff
@@ -182,9 +182,29 @@ def create(self, path):
 
     def save_config(self, path, config):
         config_path = os.path.join(path, 'config')
+        old_config_path = os.path.join(path, 'config.old')
+
+        if os.path.isfile(old_config_path):
+            logger.warning("Old config file not securely erased on previous config update")
+            self.secure_erase_config(old_config_path)
+
+        if os.path.isfile(config_path):
+            os.link(config_path, old_config_path)
```

Comment:
hmm, does this add a new requirement for the repo filesystem "must support hardlinks" that we did not have before?